lib/std/progress.zig

```diff
@@ -138,10 +142,10 @@ pub const Progress = struct {
             // restore the cursor position by moving the cursor
             // `columns_written` cells to the left, then clear the rest of the
             // line
-            if (std.builtin.os.tag != .windows) {
+            if (self.supports_ansi_escape_codes) {
                 end += (std.fmt.bufPrint(self.output_buffer[end..], "\x1b[{}D", .{self.columns_written}) catch unreachable).len;
                 end += (std.fmt.bufPrint(self.output_buffer[end..], "\x1b[0K", .{}) catch unreachable).len;
-            } else {
+            } else if (std.builtin.os.tag == .windows) {
```

Comment:
Can you add an `else unreachable`? For every other case `self.terminal` is `null`